state_machine/build.sh

```diff
@@ -1,4 +1,17 @@
 # $1 is driver, $2 is application
+
+if [ -z "$1" ]
+then
+    echo "No driver specified"
+    exit 1
+fi
+
+if [ -z "$2" ]
+then
+    echo "No application specified"
+    exit 1
+fi
+
 rm -rf build
```

Comment:
Title says make build script not rm -rf, but this command is still here? I think the changes made before this line are great, but we still recursively remove files.

Reply:
`rm -rf /`, for some reason it would try to remove everything on your computer if you didn't supply arguments.
Not sure why tbh.

Reply:
But yeah the goal was to prevent issues from not passing parameters to this script